### PR TITLE
Emergency fixes to fix a bunch of fatal errors

### PR DESCRIPTION
**Pull request by annando**
This PR fixes a bunch of fatal errors that occur during the transmitting or receiving of posts and some other background tasks.

They can be spotted by looking into the log file for entries with `Uncaught Exception`.

### CODE REVIEW

File: src/Protocol/Email.php [hunk 1]
```diff
@@ -571,7 +572,7 @@ private static function removeGPG(string $message): string
 	 * @param string $message Unfiltered message
 	 * @return string Message with no signature
 	 */
-	private static function removeSig(string $message): string
+	private static function removeSig(string $message): array
```
Comment (Quix0r) [inline src/Protocol/Email.php:575]: If this method does return `array`, please double-check and update `@return` documentation, or else they are not the same.

File: src/Util/XML.php [hunk 2]
```diff
@@ -142,7 +143,7 @@ public static function fromArray(array $array, &$xml, bool $remove_header = fals
 	 */
 	public static function copy(&$source, &$target, string $elementname)
 	{
-		if (count($source->children()) == 0) {
+		if (is_string($source) && count($source->children()) == 0) {
```
Comment (Quix0r) [inline src/Util/XML.php]: `is_string()` and `$source->children()` isn't compatible. `$foo->bar()` means object. Can you please double-check this?

File: src/Protocol/Diaspora.php [hunk 3]
```diff
@@ -4006,12 +4007,12 @@ private static function createProfileData(int $uid): array
...
 	 *
-	 * @param int  $uid    The user id
-	 * @param bool $recips optional, default false
+	 * @param int   $uid    The user id
+	 * @param array $recips optional, default empty array
 	 * @return void
 	 * @throws \Exception
 	 */
-	public static function sendProfile(int $uid, bool $recips = false)
+	public static function sendProfile(int $uid, array $recips = [])
```
Comment (Quix0r) [inline src/Protocol/Diaspora.php:4015]: Have you checked all occurrences?

**Review by Quix0r — CHANGES_REQUESTED** [review]
Please check my review comments!

File: src/Protocol/Diaspora.php (hunk 3 above)
Comment (annando) [reply src/Protocol/Diaspora.php:4015]: I had a look into the function that clearly uses an array. Also my IDE threw a lot of warnings at me, when I opened that file.

File: src/Util/XML.php (hunk 2 above)
Comment (annando) [reply src/Util/XML.php]: AFAIK the function is called with both a string and an object.

Comment (Quix0r) [reply src/Util/XML.php]: But here `is_string($source) && count($source->children())` doesn't make logically sense. You cannot access a method on a string (first check `is_string()` is true).

**Review by Quix0r — APPROVED** [review]

**Pull request merged**